docs/extensions.md

```diff
@@ -65,3 +65,51 @@ Enabling the feature would automatically infer `Some(x)` if `x` is given. In thi
     value: 5,
 )
 ```
+
+# enum_repr
+
+You can add this extension by adding the following attribute at the top of your RON document:
+
+`#![enable(enum_repr)]`
+
+This feature enables RON to declare unitary enums with numerical representations,
+There are a number of limitations to this feature:
+
+```ron
+#[repr(u8)]
+enum Utensils {
```

Comment:
I'm concerned about this view of the file now. Today, it's just the data. With this change, it becomes a mix of types and data.
I don't think we should mix them. There is no reason why the type declarations have to be specified in the same RON file. I.e. this could be done as a feature of RON library instead of being a RON format extension. For example, we could let the user provide a separate file with ".rs" extension listing those types to be used in RON.
Note: the "rs" extension is because your syntax is 100% a subset of Rust, and it's nothing like RON-described data.

Would you be open to reshaping your feature in this way?

Reply:
Seems like a totally reasonable concern, I believe that doing it as you suggest also probably allows for cleaner the lifetimes associated with the lookup tables i.e. they wouldn't need to be  thread locals any longer.

I want to have a think on it for a day or so, before diving into implementing it though.

Reply:
So, I've done an attempt at splitting up the parsers,
It is by no means a nice patch series to look review at this point.
I'll do a pass cleaning it up once we get the details nailed down.

While working on this I noticed that it is possible to split the parser and have "combined type declarations and ron data",
so they can be parsed either separately, or the types can be *prepended* onto a RON data.

The key to this is that the following is invalid rust since `#![...]` must be at the beginning of a block.

```
enum Foo{}
#![enable(...)]
```

Given the way the parser worked, I figured was actually *more* work to avoid this,
However let me know if you aren't comfortable with that being a thing.

I just wanted to throw it out there as a potential compromise,
since I believe if you wanted to extract Ron code blocks from Markdown,
performing some computed output being able to parse a combined string is going to be much easier with enums inline instead of trying to and trying to correlate
a .rs code block with a .ron one.

Let me know if you still aren't comfortable with that.

Reply:
The problem with `enum` declarations in RON is that they are simply too different from the rest. They have different syntax (requiring a different parser), they mess up the extension declaration, like you mentioned:
```
enum Foo{}
#![enable(...)]
```
If we move them to a separate file, it still seems to me that it would solve all the issues:
  1. actual RON files don't even need to know about it! they don't need `#![enable()]` it, there is no need to specify this as a format extension.
  2. the parser code can be completely different. Heck, you can use one of the existing Rust parsers for your definitions, and extra dependencies are fine because you are rolling this as an optional cargo feature anyway
  3. the definition `.rs` is automatically highlighted with Rust syntax in your editor, we don't need to break our heads trying to add support to the extension to our syntax highlighters

Reply:
Alright, well it sounds like it isn't going to particularly fit my needs, I do need the ability to parse these as a single stream, due to embedding in another format which isn't well equipped to correlate the separated and pass the lookup table between the .rs parser and the .ron parser.

As such I think i'll close this and leave the branch up, It will be easier to round out some of the rougher edges with an ad-hoc format by not trying to involve existing formats and syntax anyways.

Reply:
> (sorry for late feedback!)



> (sorry for late feedback!)

No problem, thank you for taking the time to review though!